### DISCUSSION TITLE
bug: Constant errors when using Rust extra

### DISCUSSION THREAD

**Discussion by osenvosem** [LazyVim/LazyVim#4195] [Q&A]
### Did you check docs and existing issues?

- [X] I have read all the LazyVim docs
- [X] I have updated the plugin to the latest version before submitting this issue
- [X] I have searched the existing issues of LazyVim
- [X] I have searched the existing issues of plugins related to this issue

### Neovim version (nvim -v)

0.10.1

### Operating system/version

MacOS 13.6

### Describe the bug

Huge number of errors appear when writing and moving Rust code. Whether it is Rust stable or Rust nightly (it seem with nightly there are even more errors).
No configuration for the Rust extra plugins changed.
Tried to ask in Discussions in this repo and in the rustecian.nvim repo with no luck.

<img width="1402" alt="350112753-2ea2a74b-9be3-4a47-a8c4-7c978d782b65" src="https://github.com/user-attachments/assets/11621b00-9717-490e-9151-b4901daefcdf">

### Steps To Reproduce

Install Rust Extra and try

### Expected Behavior

No errors

### Repro

```lua
vim.env.LAZY_STDPATH = ".repro"
load(vim.fn.system("curl -s https://raw.githubusercontent.com/folke/lazy.nvim/main/bootstrap.lua"))()

require("lazy.minit").repro({
  spec = {
    { "LazyVim/LazyVim", import = "lazyvim.plugins" },
    -- add any other plugins here
  },
})
```

**Comment by dpetka2001** [comment]
How did you install Rust? Via `rustup` or your distro's package manager? I've seen several issues at [rust-analyzer repo](https://github.com/rust-lang/rust-analyzer/issues?q=sort%3Aupdated-desc+is%3Aissue+is%3Aopen+overly+long+loop+turn) as well. Your best bet is to ask over there or `rustaceanvim`'s repo, as LazyVim just uses the tools provided.

**Comment by osenvosem** [comment]
> How did you install Rust? Via `rustup` or your distro's package manager? I've seen several issues at [rust-analyzer repo](https://github.com/rust-lang/rust-analyzer/issues?q=sort%3Aupdated-desc+is%3Aissue+is%3Aopen+overly+long+loop+turn) as well. Your best bet is to ask over there or `rustaceanvim`'s repo, as LazyVim just uses the tools provided.

I installed Rust via Homebrew. There is no such issues with any other editor, I used RustRover, VSCode and Zed.
As I mentioned before, I asked in the rustaceanvim's repo https://github.com/mrcjkb/rustaceanvim/discussions/454 and it seems like the issue is on LazyVim side.

**Comment by dpetka2001** [comment]
The maintainer of `rustaceanvim` suggested that this may be due to a `rust-analyzer` debug build. Unfortunately, I don't do any Rust to know better about this. But could you try uninstalling Rust via Homebrew and install it via `rustup` as per [this comment](https://github.com/rust-lang/rust-analyzer/issues/3959#issuecomment-1058992018) from the issue that `rustaceanvim`'s maintainer suggested?

**Comment by osenvosem** [comment]
> The maintainer of `rustaceanvim` suggested that this may be due to a `rust-analyzer` debug build. Unfortunately, I don't do any Rust to know better about this. But could you try uninstalling Rust via Homebrew and install it via `rustup` as per [this comment](https://github.com/rust-lang/rust-analyzer/issues/3959#issuecomment-1058992018) from the issue that `rustaceanvim`'s maintainer suggested?

Sorry, I was wrong. Rust is actually installed using rustup not homebrew. And again no issues in any other editor.

**Reply by dpetka2001** [reply]
Can you try with a clean Rust project (`cargo init`)? Does it still happen? If not, it has something to do with your current project and it would be best if you posted a question in `rust-analyzer` github repo. I can't reproduce this with Rust Extra and a simple clean Rust project. Again, I don't do Rust and I just tried a simple `hello world` program.
